cartridges/int_klaviyo_core/cartridge/scripts/klaviyo/utils.js

```diff
@@ -0,0 +1,247 @@
+    bonusProductData.priceValue = lineItemObj.adjustedPrice.value;
+
+    return bonusProductData;
+}
+
+
+// helper function to consider promos & set Price and Original Pride properties on a returned object.
+// Used in order level events: 'Started Checkout' and 'Order Confirmation'.
+function priceCheck (lineItemObj, basketProdObj) {
+    var priceModel = basketProdObj ? basketProdObj.getPriceModel() : null;
+    var priceBook = priceModel ? getRootPriceBook(priceModel.priceInfo.priceBook) : null;
+    var priceBookPrice = priceBook && priceModel ? priceModel.getPriceBookPrice(priceBook.ID) : null;
+    var priceData = {};
+
+    var adjustedPromoPrice = lineItemObj && lineItemObj.adjustedPrice < priceBookPrice ? StringUtils.formatMoney(dw.value.Money( lineItemObj.adjustedPrice.value, session.getCurrency().getCurrencyCode() )) : null;
+    if (adjustedPromoPrice) {
+        priceData.purchasePrice = StringUtils.formatMoney(dw.value.Money( lineItemObj.adjustedPrice.value, session.getCurrency().getCurrencyCode() ));
+        priceData.purchasePriceValue = lineItemObj.adjustedPrice.value;
+        priceData.originalPrice = priceBookPrice ? StringUtils.formatMoney(dw.value.Money( priceBookPrice.value, session.getCurrency().getCurrencyCode() )) : StringUtils.formatMoney(dw.value.Money( basketProdObj.getPriceModel().getPrice().value, session.getCurrency().getCurrencyCode() ));
+        priceData.originalPriceValue = priceBookPrice.value;
+    } else {
+        priceData.purchasePrice = lineItemObj ? StringUtils.formatMoney(dw.value.Money( lineItemObj.price.value, session.getCurrency().getCurrencyCode() )) : null;
+        priceData.purchasePriceValue = lineItemObj ? lineItemObj.price.value : null;
+        priceData.originalPrice = basketProdObj ? StringUtils.formatMoney(dw.value.Money( basketProdObj.getPriceModel().getPrice().value, session.getCurrency().getCurrencyCode() )) : null;
+        priceData.originalPriceValue = basketProdObj.getPriceModel().getPrice().value;
+    }
+
+    return priceData
+}
+
+
+/**
+ * Return root price book for a given price book
+ * @param {dw.catalog.PriceBook} priceBook - Provided price book
+ * @returns {dw.catalog.PriceBook} root price book
+ */
+function getRootPriceBook(priceBook) {
+    var rootPriceBook = priceBook;
+    while (rootPriceBook.parentPriceBook) {
+        rootPriceBook = rootPriceBook.parentPriceBook;
+    }
+    return rootPriceBook;
+}
+
+
+function trackEvent(exchangeID, data, event, customerEmail) {
+
+    var requestBody = {};
+    var resultObj = {};
+    var logger = Logger.getLogger('Klaviyo', 'Klaviyo.core utils.js - trackEvent()');
+
+    if (klaviyoServices.KlaviyoEventService == null || empty(exchangeID)) {
+        logger.error('trackEvent() failed - KlaviyoEventService or exchange_id is null.  exchange_id: ' + exchangeID + '.');
+        return;
+    }
+
+    /* IMPORTANT:
```

Comment:
This logic seems specific to Added to Cart - would it make more sense in the getData method?

Reply:
we would actually need to add it to the controllers for Add to Cart, ~line 35 in SFRA's Cart.js and ~line 290 in SG's Cart.js (in both cases just before the call to klaviyoUtils.trackEvent.  at the time my thoughts ran along the lines of keeping the code in the controllers as simple and standard across controllers as possible, and also keeping the funk of the site owner needing to understand the importance of setting both the "Send Added to Cart as Add to Cart" and "Send Events with Salesforce Label" adjacent in the code itself... but all that said you are correct that this very specific to that one event type.  let me know if you'd like this shifted to the controller level and we'll do it.

Reply:
nix that.  in thinking on this further, i now realize that we have what could be considered a bug here.  as is, we currently pass the event type at two locations in the data - one in the object built by the getData functions, and another in the "meta" or top-level data that gets wrapped around it in trackEvent.  with the code that exists today we are only correcting this to "Add to Cart" at the meta level in trackEvent, so there's a discrepancy between that value and the one in the getData function.  we probably need to consider referencing the value set in the getData object to set that value at the meta level in trackEvent, but it's important to understand that this would represent a signature change to the trackEvent call and would require testing across all events to ensure we didn't break anything in the process.